keps/draft-20190516-KUDO-package-manager.md

```diff
@@ -0,0 +1,177 @@
+---
+kep-number: draft-05162019
+title: KUDO Package Manager
+authors:
+  - "@fabianbaier"
+owners:
+  - TBD
+  - "@fabianbaier"
+editor: TBD
+creation-date: 2019-05-16
+status: provisional
+---
+
+# package-manager-for-distributing-kudo-packages
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+ * [Goals](#goals)
+ * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+ * [User Stories](#user-stories)
+    * [Framework Developer](#framework-developer)
+    * [Cluster Administrator](#cluster-administrator)
+ * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+ * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Infrastructure Needed](#infrastructure-needed)
+
+## Summary
+
+As more and more developers seed our `kubebuilder/frameworks` repository with incredibly important Frameworks, there needs to be a provided structure that talks about how this repository is organized and made available. The underlying and agreed on structure also indirectly impacts or tangents to other KEPs, e.g. [KEP-0003 CLI](https://github.com/kudobuilder/kudo/blob/master/keps/0003-kep-cli.md), [KEP-0007 CLI Gernation](https://github.com/kudobuilder/kudo/blob/master/keps/0007-cli-generation.md) or [KEP-0008 Framework Testing](https://github.com/kudobuilder/kudo/blob/master/keps/0008-framework-testing.md).
+
+Overall, this KEP should capture how we plan to provide a great user experience of installing Frameworks with KUDO. Attention should be split to:
+
+* How the Package structure looks like and dictates the overall repository structure
+* What tools should be used to achieve this structure
+* How are packages are distributed to the user
+
+## Motivation
+
+There are multiple package managers and ideas out there and we need to decide which concept will work the best for KUDO in its current state. This is not an Install CLI subcategory KEP, but of course we need to think about the repercussions of decisions made in this KEP. There are multiple stakeholders directly being impacted by decisions made here, but this also paves the way to think about new problems we haven't articulated yet (e.g. how to provide KUDO Frameworks in air-gapped clusters, verification of Frameworks and their tarbundles, etc.).
+
+### Goals
+
+Specific goals are:
+* Coming up with a structure for a Package
+* Defining the overall `kubebuilder/framework` repository structure
+* Having a solution on how to distribute this structure
+* Having this platform-agnostic so Packages can be distributed across multiple solutions
```

Comment:
I'd rephrase this like this: The solution should be platform-agnostic, so that Packages can be used on multiple platforms/

Although it isn't very clear what platforms/solutions we're talking about. Different orchestrators/OS/k8s tools?

Reply:
Good point! Something we should think about in future releases.

This was meant more towards making sure however we distribute the package, here as `.tgz`, won't break on different platforms (OS).